components/drafts/draft_title/index.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getUser, getUserIdsInChannels} from 'mattermost-redux/selectors/entities/users';
+import {getUserIdFromChannelName} from 'mattermost-redux/utils/channel_utils';
+import {getTeammateNameDisplaySetting} from 'mattermost-redux/selectors/entities/preferences';
+import {displayUsername} from 'mattermost-redux/utils/user_utils';
+import {Constants} from 'utils/constants';
+
+import {Channel} from 'mattermost-redux/types/channels';
+import {GlobalState} from 'types/store';
+
+import DraftTitle from './draft_title';
+
+type OwnProps = {
+    channel: Channel;
+    userId: string;
+};
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const {channel, userId} = ownProps;
+
+    let channelName = channel.display_name;
+    let teammateId;
+    let teammate;
+    let membersCount;
+
+    if (channel.type === Constants.DM_CHANNEL) {
+        teammateId = getUserIdFromChannelName(userId, channel.name);
+        teammate = getUser(state, teammateId);
+        channelName = displayUsername(teammate, getTeammateNameDisplaySetting(state));
+    }
+
+    if (channel.type === Constants.GM_CHANNEL) {
+        const memberIds = getUserIdsInChannels(state);
+
+        membersCount = 0;
+        if (memberIds && memberIds[ownProps.channel.id]) {
```

Comment:
You already have the channel in the current context (line 23), so no need to take it back from ownProps.

Reply:
Same on next line.